### PR TITLE
COMMUNITY-80 | Add a new stage in the build_and_test script to publish connector into our internal npm server when a new tag is pushed

### PR DESCRIPTION
**Pull request by ManuelJNunez**
In this PR I added a new stage to publish connector into our internal server when a new tag with format `v\d+\.\d+\.\d+` is pushed. That new stage will call a bash script that I implemented for that purpose.

### CODE REVIEW

**Comment by ManuelJNunez** [comment]
[ ] Pending on https://github.com/rticommunity/rticonnextdds-connector/pull/95

File: resources/scripts/publish.sh [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+libraries=(**/lib/*/*.so)
+
+if [ ${#libraries[@]} -eq 0 ]; then
+    echo "Error: No libraries were found, download them before using this script."
+    exit 1
+fi
+
+build_id=$(strings ${libraries[0]} | grep -o 'BUILD_.*')
+modified_build_id=${build_id//_/\-}
+version_to_publish=$package_version-$modified_build_id
```
Comment (Alxe) [inline resources/scripts/publish.sh:16]: We need to make sure that this script fits both internal and external uploads. For external uploads, we don't want to modify the version of the package.

File: ci_config.yaml [hunk 2]
```diff
@@ -3,3 +3,4 @@ node_versions:
   - "19"
   - "20"
   - "lts"
+publish_version: "lts"
```
Comment (Alxe) [inline ci_config.yaml:6]:
```suggestion
publish_version: "20"
```

While we're most likely going to use the LTS for publication, I think it's best we make that choice explicitly.

File: resources/jenkins/build_and_test.groovy [hunk 3]
```diff
@@ -140,6 +150,33 @@ pipeline {
...
+            when {
+                tag pattern: /v\d+\.\d+\.\d+/, comparator: "REGEXP"
+            }
+
+            steps {
+                withCredentials([
+                    string(credentialsId: 'npm-registry', variable: 'NPM_RESGISTRY'),
+                    string(credentialsId: 'npm-token', variable: 'NPM_TOKEN')
+                ]) {
+                    dir("${env.WORKSPACE}/${CI_CONFIG['publish_version']}") {
```
Comment (Alxe) [inline resources/jenkins/build_and_test.groovy:173]: Is `CI_CONFIG` valid in this scenario? Based on previous code, this is only true when the node version is not predefined in the job name.

**Review by Alxe — CHANGES_REQUESTED** [review]

**Review by Alxe — APPROVED** [review]

**Pull request merged**